velox/exec/Merge.cpp

```diff
@@ -123,102 +101,276 @@ BlockingReason Merge::isBlocked(ContinueFuture* future) {
   return BlockingReason::kWaitForProducer;
 }
 
-void Merge::startSources() {
-  VELOX_CHECK_LE(numStartedSources_, sources_.size());
-  // Start the merge source once.
+bool Merge::isFinished() {
+  return finished_;
+}
+
+void Merge::setupOutputSpiller() {
+  VELOX_CHECK(spillConfig_.has_value());
+  mergeOutputSpiller_ = std::make_unique<MergeSpiller>(
+      outputType_,
+      std::nullopt,
+      HashBitRange{},
+      sortingKeys_,
+      &spillConfig_.value(),
+      &spillStats_);
+}
+
+void Merge::spill() {
+  VELOX_CHECK_NOT_NULL(output_);
+  setupOutputSpiller();
+  numSpilledRows_ += output_->size();
```

Comment:
I think we need to check if output_ is null or not here?

Reply:
The output_ must not be null if the spill() is called, so I added `VLOX_CHECK_NOT_NULL(output_)`

Reply:
I guess it might be possible that SourceMerger::getOutput return nullptr with atEnd = true for some corner case that all the sources to merge are empty?